Content.Server/Sports/BaseballBatSystem.cs

```diff
@@ -0,0 +1,85 @@
+using Content.Server.Sports.Components;
+using Content.Server.Weapon.Melee.Components;
+using Content.Server.Weapon.Ranged.Systems;
+using Content.Shared.Item;
+using Content.Shared.Projectiles;
+using Content.Shared.Throwing;
+using Content.Shared.Weapons.Melee;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+using Content.Shared.Weapons.Ranged;
+using Content.Shared.Weapons.Ranged.Systems;
+using Robust.Shared.Timing;
+
+
+namespace Content.Server.Sports
+{
+    public sealed class BaseballBatSystem : EntitySystem
+    {
+        [Dependency] private readonly ThrowingSystem _throwingSystem = default!;
+        [Dependency] private readonly EntityLookupSystem _entityLookupSystem = default!;
+        [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly GunSystem _gunSystem = default!;
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<BaseballBatComponent, WideAttackEvent>(OnBatAttack);
+        }
+
+        public void OnBatAttack(EntityUid uid, BaseballBatComponent component, WideAttackEvent args)
+        {
+            args.Handled = true;
+
+            if (!EntityManager.TryGetComponent<MeleeWeaponComponent>(component.Owner, out var meleeWeaponComponent))
```

Comment:
You can just use the proxy methods in `EntitySystem.Proxy` instead of the longer form, the same applies to all the try and get components in the rest of the files

Reply:
Didn't know this existed, thanks